tests/Unit/IO/Test_StellarCollapseEos.cpp

```diff
@@ -0,0 +1,347 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include <algorithm>
+#include <cstddef>
+#include <string>
+#include <vector>
+
+#include "DataStructures/BoostMultiArray.hpp"
+#include "IO/H5/AccessType.hpp"
+#include "IO/H5/File.hpp"
+#include "IO/H5/Header.hpp"  // IWYU pragma: keep
+#include "IO/H5/StellarCollapseEos.hpp"
+#include "Informer/InfoFromBuild.hpp"
+#include "tests/Unit/TestingFramework.hpp"
+
+namespace {
+
+// For 1D arrays the numbers (in, for example, logrho_expected) are the first
+// and second numbers in the corresponding dataset. For non-zero 3D arrays with
+// no repeating numbers they are the first 2x2x2 block
+// (all elements with indices [0][0][0] -> [1][1][1])
+// For 3D arrays with every element equal to zero (or with repeating numbers),
+// they are random numbers in the range [-10.0, 10.0]
+void test_tabulated(const std::string& file_path,
+                    const std::string& subgroup_path) noexcept {
+  h5::H5File<h5::AccessType::ReadOnly> sample_file(file_path);
+  const auto& sample_data =
+      sample_file.get<h5::StellarCollapseEos>(subgroup_path);
+
+  // Check that the group /sample_data can be opened and read from
```

Comment:
The comment and code don't really complement each other. If I didn't know what you were trying to say I'd have no clue. You should mention that `/sample_data` has the extra field `test_data` which you read to verify you are not in `/`

Reply:
I don't really like the passive-aggressive tone in some of your code review comments. Your second sentence is just unnecessary.